RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelRawToClusterGPUKernel.cu

```diff
@@ -451,56 +450,86 @@ namespace pixelgpudetails {
     }  // end of loop (gIndex < end)
 
   }  // end of Raw to Digi kernel
-
+  template <bool isPhase2>
   __global__ void fillHitsModuleStart(uint32_t const *__restrict__ clusInModule,
                                       uint32_t *__restrict__ moduleStart,
                                       uint32_t const *__restrict__ nModules,
                                       uint32_t *__restrict__ nModules_Clusters) {
-    assert(gpuClustering::maxNumModules < 2048);  // easy to extend at least till 32*1024
+    constexpr int nMaxModules = isPhase2 ? phase2PixelTopology::numberOfModules : phase1PixelTopology::numberOfModules;
+    constexpr int startBPIX2 = isPhase2 ? phase2PixelTopology::layerStart[1] : phase1PixelTopology::layerStart[1];
+
+    assert(nMaxModules < phase2PixelTopology::numberOfModules);
```

Comment:
Are all these `assert`'s really needed? I mean, they are constants defined in GeometryBuilder/interface/pixelTopology.h: I think you can double check now that they are coded correctly, and then give them for granted at run time. Or am I missing something?